packages/cardpay-reward-programs/tests/rules/test_min_other_merchants_paid.py

```diff
@@ -50,60 +46,48 @@ def rule(request):
 
 class TestMinOtherMerchantsPaidSingle:
     @pytest.mark.parametrize(
-        "rule,ans",
+        "rule,summary",
         zip(
             itertools.product(payment_cycle_length_ls, min_other_merchants_ls),
-            ans_ls,
+            summaries,
         ),
         indirect=["rule"],
     )
-    def test_run(self, rule, ans, indexed_data):
-        df_hash, summary = ans
+    def test_run(self, rule, summary, indexed_data):
         start_block = 24000000
         end_block = 26000000
-        df = rule.run(start_block, end_block)
-        payment_list = rule.df_to_payment_list(df)
-        h = hashlib.sha256(pd.util.hash_pandas_object(df, index=True).values).hexdigest()
+        rule.payment_cycle_length = end_block - start_block
+        payment_list = rule.run(end_block, "0x0")
         computed_summary = rule.get_summary(payment_list)
         assert computed_summary["total_reward"][0] == summary["total_reward"]
         assert computed_summary["unique_payee"][0] == summary["unique_payee"]
-        assert h == df_hash
 
 
-multiple_df_hashes = [
```

Comment:
These hashes were making the test very brittle, and quite hard to interpret failures - I'm not sure how best to update these either.

Reply:
Yea strongly agree. It was working initially then it was hell once I started changing the code.